tests/src/OneLogin/saml2_tests/idp_metadata_parser_test.py

```diff
@@ -51,7 +51,7 @@ def testParseRemote(self):
 
         data = OneLogin_Saml2_IdPMetadataParser.parse_remote('https://www.testshib.org/metadata/testshib-providers.xml')
         self.assertTrue(data is not None and data is not {})
-        expected_data = {'sp': {'NameIDFormat': 'urn:mace:shibboleth:1.0:nameIdentifier'}, 'idp': {'singleSignOnService': {'url': 'https://idp.testshib.org/idp/profile/SAML2/Redirect/SSO'}, 'entityId': 'https://idp.testshib.org/idp/shibboleth'}}
+        expected_data = {'sp': {'NameIDFormat': 'urn:mace:shibboleth:1.0:nameIdentifier'}, 'idp': {'singleSignOnService': {'url': 'https://idp.testshib.org/idp/profile/SAML2/Redirect/SSO', 'binding': 'urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect'}, 'entityId': 'https://idp.testshib.org/idp/shibboleth'}}
         self.assertEqual(expected_data, data)
 
     def testParse(self):
```

Comment:
You can add different IdP metadata endpoints to the test suits and try to parse them.
I'm ok with the test that check the settings merge. I want to cover all the "parse" scenarios.

IdP metadata with various SSO endpoints and try to extract offering different binding and see if the extracted value works.

Same for SLO endpoints (taking in mind that SLO endpoint is not required in a metadata XML)